### PR TITLE
Extracting game information 

### PR DESCRIPTION
**Pull request by gawsalya**
No tests written yet, purely for initial comments

### CODE REVIEW

File: live_pipeline/test.csv
Comment (DMcCallion) [inline live_pipeline/test.csv:1]: We probably want *.csv in our gitignore

File: live_pipeline/extract_games.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+from urllib.request import urlopen
```
Comment (DMcCallion) [inline live_pipeline/extract_games.py]: Missing module docstring

File: live_pipeline/extract_games.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+from bs4 import BeautifulSoup
+import pandas as pd
+import requests
```
Comment (DMcCallion) [inline live_pipeline/extract_games.py:4]: alphabetise imports

File: live_pipeline/extract_games.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    response = data['publishers']
+    for publisher in response:
+        publishers.append(publisher)
+    return publishers
+
+
+if __name__ == "__main__":
+
+    website = get_html(
+        "https://store.steampowered.com/search/?sort_by=Released_DESC&category1=998&supportedlang=english&ndl=1")
```
Comment (DMcCallion) [inline live_pipeline/extract_games.py]: The loose URLs should probably be global variables or .env variables

File: live_pipeline/extract_games.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+        response = request.json()[game['app_id']]['data']
+        compatible_systems = system_requirements(response)
+        game.update(compatible_systems)
+        steam_genres = get_genre_from_steam(response)
+        game['genres'] = steam_genres
+        developer = get_developer_name(response)
+        game['developers'] = developer
+        publisher = get_publisher_name(response)
+        game['publishers'] = publisher
```
Comment (DMcCallion) [inline live_pipeline/extract_games.py]: Doing a lot in main here: This could be its own function

File: live_pipeline/extract_games.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+import pandas as pd
+import requests
+
+
+def get_html(url):
+    """function to open the url and get the information"""
+    page = urlopen(url)
+    html_bytes = page.read()
+    html = html_bytes.decode("utf_8")
+    return html
```
Comment (rhKashif) [inline live_pipeline/extract_games.py]: Refactor: We could implement some error handling here using a try-except block to ensure we don't crash the code if there is something wrong with opening the url? E.g. 

```
def get_html(url):
    try:
        response = requests.get(url)
        response.raise_for_status()
        html = response.text
        return html
    except Exception as e:
        print(f"Error: {e}")
        return None
```
  
If you can implement a more specific exception type for the error raised even better!

**Review by DMcCallion — COMMENTED** [review]
Looks good to merge after making those changes

File: live_pipeline/extract_games.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
+from urllib.request import urlopen
+from bs4 import BeautifulSoup
+import pandas as pd
+import requests
+
+
+def get_html(url):
```
Comment (rhKashif) [inline live_pipeline/extract_games.py]: Nit Pick: Type hint for arguments and expected returns.

File: live_pipeline/extract_games.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    """function to open the url and get the information"""
```
Comment (rhKashif) [inline live_pipeline/extract_games.py:12]: Add Args and Returns for the function to the doc string

File: live_pipeline/extract_games.py (hunk 1 above)
Comment (rhKashif) [inline live_pipeline/extract_games.py]: General Comment: Add args and returns for the function to the doc string

File: live_pipeline/extract_games.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    game['publishers'] = publisher
+
+
+if __name__ == "__main__":
+
+    website = get_html(RELEASE_WEBSITE)
+    all_recent_games = parse_app_id_bs(website)
+
+    for game in all_recent_games:
+        update_game_information(game["app_id"])
```
Comment (rhKashif) [inline live_pipeline/extract_games.py:129]: Can we put this into a function?

**Comment by angelikavilde** [comment]
Looks good to me!

**Pull request merged**